### PR TITLE
Implement percentile

### PR DESCRIPTION
**Pull request by hvy**
Implemented percentile for CuPy. 

See https://docs.scipy.org/doc/numpy-dev/reference/generated/numpy.percentile.html

### CODE REVIEW

File: cupy/statistics/order.py [hunk 1]
```diff
@@ -113,4 +114,114 @@ def nanmax(a, axis=None, out=None, keepdims=False):
 # TODO(okuta): Implement ptp
 
 
-# TODO(okuta): Implement percentile
+def percentile(a, q, axis=None, out=None, interpolation='linear',
+               keepdims=False):
+    """Compute the qth percentile of the data along the specified axis.
```
Comment (unnonouno) [inline cupy/statistics/order.py]: Compute -> Computes
qth -> q-th (?)

File: cupy/statistics/order.py [hunk 2]
```diff
@@ -113,4 +114,114 @@ def nanmax(a, axis=None, out=None, keepdims=False):
...
+
+    if interpolation == 'lower':
+        indices = cupy.floor(indices).astype(cupy.int32)
+    elif interpolation == 'higher':
+        indices = cupy.ceil(indices).astype(cupy.int32)
+    elif interpolation == 'midpoint':
+        indices = 0.5 * (cupy.floor(indices) + cupy.ceil(indices))
+    elif interpolation == 'nearest':
+        # TODO(hvy): Implement nearest using around
+        raise NotImplementedError()
```
Comment (unnonouno) [inline cupy/statistics/order.py]: `NotImplementedError` is used when an abstract class does not have an implementation. Use `ValueError`.
https://docs.python.org/3.6/library/exceptions.html#NotImplementedError

File: cupy/statistics/order.py [hunk 3]
```diff
@@ -113,4 +114,114 @@ def nanmax(a, axis=None, out=None, keepdims=False):
...
+    elif interpolation == 'linear':
+        pass
+    else:
+        msg = 'Unexpected interpolation method.\n' \
+            'Actual: {0}'.format(interpolation)
+        raise ValueError(msg)
```
Comment (unnonouno) [inline cupy/statistics/order.py]: test this case

File: cupy/statistics/order.py (hunk 1 above)
Comment (hvy) [reply cupy/statistics/order.py]: qth was actually taken from the NumPy docs, but this looks better. Thank you, fixed.

File: cupy/statistics/order.py (hunk 2 above)
Comment (hvy) [reply cupy/statistics/order.py]: Fixed.

File: cupy/statistics/order.py (hunk 3 above)
Comment (hvy) [reply cupy/statistics/order.py]: Added a test.

**Comment by hvy** [comment]
@unnonouno Thank you for the review. I made some changes to the code accordingly.

File: cupy/statistics/order.py [hunk 4]
```diff
@@ -113,4 +114,114 @@ def nanmax(a, axis=None, out=None, keepdims=False):
...
+
+    if interpolation == 'lower':
+        indices = cupy.floor(indices).astype(cupy.int32)
+    elif interpolation == 'higher':
+        indices = cupy.ceil(indices).astype(cupy.int32)
+    elif interpolation == 'midpoint':
+        indices = 0.5 * (cupy.floor(indices) + cupy.ceil(indices))
+    elif interpolation == 'nearest':
+        # TODO(hvy): Implement nearest using around
+        raise ValueError()
```
Comment (unnonouno) [inline cupy/statistics/order.py]: Add an error message such as "'nearest' is not supported yet"

Comment (hvy) [reply cupy/statistics/order.py]: Fixed this error message and also made the other error more verbose.

**Comment by hvy** [comment]
@unnonouno san, I fixed the error message accordingly. It is ready for review.

**Comment by unnonouno** [comment]
LGTM

**Pull request merged**